vmm/src/lib.rs

```diff
@@ -781,30 +797,29 @@ impl Vmm {
         drive_id: &str,
         disk_image: File,
     ) -> result::Result<(), DriveError> {
-        if let Some(device_idx) = self.drive_handler_id_map.get(drive_id) {
-            match self.epoll_context.get_device_handler(*device_idx) {
-                Ok(handler) => {
-                    match handler.handle_event(
-                        virtio::block::FS_UPDATE_EVENT,
-                        *device_idx as u32,
-                        EpollHandlerPayload::DrivePayload(disk_image),
-                    ) {
-                        Err(devices::Error::PayloadExpected) => {
-                            panic!("Received update disk image event with empty payload.")
-                        }
-                        Err(devices::Error::UnknownEvent { device, event }) => {
-                            panic!("Unknown event: {:?} {:?}", device, event)
-                        }
-                        _ => Ok(()),
+        match self
+            .epoll_context
+            .get_device_handler_by_device_id(TYPE_BLOCK, drive_id)
+        {
+            Ok(handler) => {
+                match handler.handle_event(
+                    virtio::block::FS_UPDATE_EVENT,
+                    0,
```

Comment:
Looks like `event_flags: u32` param of `handle_event` is never used, let's take this chance to drop it.

Reply:
Done